app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCStatus.scala

```diff
@@ -0,0 +1,295 @@
+      case _: Accepted        => "ACCEPTED"
+      case _: Signed          => "SIGNED"
+      case _: Broadcasted     => "BROADCASTED"
+      case _: Confirmed       => "CONFIRMED"
+      case _: CloseOffered    => "CLOSE OFFERED"
+      case _: Closed          => "CLOSED"
+      case _: Claiming        => "CLAIMING"
+      case _: Claimed         => "CLAIMED"
+      case _: Penalized       => "PENALIZED"
+      case _: RemoteClaiming  => "REMOTE CLAIMING"
+      case _: RemoteClaimed   => "REMOTE CLAIMED"
+      case _: RemotePenalized => "REMOTE PENALIZED"
+      case _: Refunded        => "REFUNDED"
+    }
+  }
+
+  case class Offered(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer)
+      extends DLCStatus {
+
+    def toAccepted(accept: DLCAccept): Accepted = {
+      Accepted(eventId, isInitiator, offer, accept)
+    }
+  }
+
+  case class Accepted(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept)
+      extends AcceptedDLCStatus {
+
+    def toSigned(sign: DLCSign): Signed = {
+      Signed(eventId, isInitiator, offer, accept, sign)
+    }
+  }
+
+  case class Signed(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign)
+      extends AcceptedDLCStatus {
+
+    def toBroadcasted(fundingTx: Transaction): Broadcasted = {
+      Broadcasted(eventId, isInitiator, offer, accept, sign, fundingTx)
+    }
+
+    def toConfirmed(fundingTx: Transaction): Confirmed = {
+      toBroadcasted(fundingTx).toConfirmed
+    }
+  }
+
+  case class Broadcasted(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction)
+      extends AcceptedDLCStatus {
+
+    def toConfirmed: Confirmed = {
+      Confirmed(eventId, isInitiator, offer, accept, sign, fundingTx)
+    }
+  }
+
+  case class Confirmed(
+      eventId: Sha256DigestBE,
+      isInitiator: Boolean,
+      offer: DLCOffer,
+      accept: DLCAccept,
+      sign: DLCSign,
+      fundingTx: Transaction)
+      extends AcceptedDLCStatus {
+
+    def toCloseOffered(closeSig: DLCMutualCloseSig): CloseOffered = {
+      CloseOffered(eventId,
+                   isInitiator,
+                   offer,
+                   accept,
+                   sign,
+                   fundingTx,
+                   closeSig)
+    }
+
+    def toClaiming(
+        oracleSig: SchnorrDigitalSignature,
+        cet: Transaction): Claiming = {
+      Claiming(eventId,
+               isInitiator,
+               offer,
+               accept,
+               sign,
+               fundingTx,
+               oracleSig,
+               cet)
+    }
+
+    def toRemoteClaiming(cet: Transaction): RemoteClaiming = {
+      RemoteClaiming(eventId, isInitiator, offer, accept, sign, fundingTx, cet)
+    }
+  }
+
+  case class CloseOffered(
```

Comment:
scaladocs would be useful, so this is for the case where we are aborting and offer?